src/main/java/org/codehaus/plexus/components/io/resources/PlexusIoFileResource.java

```diff
@@ -171,16 +177,28 @@ public long getSize()
 
     public boolean isDirectory()
     {
+        if ( attributes instanceof FileAttributes )
+        {
+            return ( ( FileAttributes ) attributes ).isDirectory();
+        }
         return getFile().isDirectory();
     }
 
     public boolean isExisting()
     {
+        if ( attributes instanceof FileAttributes )
+        {
+            return true;
+        }
         return getFile().exists();
     }
 
     public boolean isFile()
     {
+        if ( attributes instanceof FileAttributes )
```

Comment:
So the issue was that `File` returns for the symlink destination and FileAttributes for the symlink? Can then just get the attributes for the destination?

Reply:
So the overall goal is to reduce the number of calls to retrieve file attributes, i.e. to cache them.  The `if ( attributes instances FileAttributes )` are here to make sure we keep the existing behavior in the non usual cases, i.e. where `attributes` is not an instance of `FileAttributes`.  This could eventually happen if people uses the factory and pass a custom attributes object to https://github.com/codehaus-plexus/plexus-io/blob/8504fa17d9331fc63b603999296d69961397f880/src/main/java/org/codehaus/plexus/components/io/resources/ResourceFactory.java#L50.  In most cases, a `FileAttributes` instance will be used.

Reply:
What I meant is why we can have single attributes. But you answer than on the other comment.